helm/nginx-ingress-controller-app/values.yaml

```diff
@@ -1,89 +1,203 @@
+
+  # configmap.worker-shutdown-timeout
+  # Maximum amount of time NGINX worker processes should give active connections to drain.
+  # This should not be higher than controller.terminationGracePeriodSeconds
   worker-shutdown-timeout: "240s"
+
+  # configmap.use-forwarded-headers
+  # If true, NGINX passes the incoming X-Forwarded-* headers to upstreams.
   use-forwarded-headers: "true"
 
+# controller
+# Contains mostly configuration values that get applied to the kubernetes
+# resources that make up the ingress controller.
 controller:
+
+  # controller.name
+  # Gets applied to most metadata.name fields in the various resources that this
+  # chart creates.
   name: nginx-ingress-controller
+
+  # controller.k8sAppLabel
+  # Gets applied to the `app` label in the metadata.labels of the various resources
+  # that this chart creates.
   k8sAppLabel: nginx-ingress-controller
 
+  # controller.replicaCount
+  # Number of initial NGINX IC Deployment replicas.
   replicaCount: 1
 
+  # controller.maxUnavailable
+  # Configures maximum number of unavailable replicas while doing a
+  # rolling upgrade.
   maxUnavailable: 1
+
+  # controller.minReadySeconds
   # minReadySeconds to avoid killing pods before we are ready
+  # Configures the minimum amount of time that a replica has to be
+  # ready before a rolling upgrade can proceed with the next replica.
   minReadySeconds: 0
 
+  # controller.configmap
+  # TODO: Explain what this is for
   configmap:
+
+    # controller.configmap.name
+    # The name that the configuration configmap will get. This is the configmap
+    # that gets filled with values from the `configmap` field in this values
+    # file.
     name: ingress-nginx
 
+  # controller.image
   image:
+
+    # controller.image.repository
     repository: giantswarm/nginx-ingress-controller
-    # when updating tag make sure to also keep appVersion in Chart.yaml in sync
+
+    # controller.image.tag
+    # When updating tag make sure to also keep appVersion in Chart.yaml in sync
     tag: 0.30.0
 
+  # controller.rbac
+  # TODO: Explain what this is for
   rbac:
 
+  # controller.role
+  # TODO: Explain what this is for
   role:
+
+    # controller.role.name
+    # The name tha the RBAC role will get.
     name: nginx-ingress-role
 
+  # controller.service
   service:
+
+    # controller.service.aws
+    # If true, create NodePort Service. Dynamically calculated during cluster creation.
     enabled: false
-    # type applies only to AWS provider:
-    # - external (default)
-    # - internal
-    # ...
+
+    # controller.service.type
+    # Applies only to clusters running on AWS.
+    # Valid values: external, internal
     type: external
+
+    # controller.service.nodePorts
     # Sets the NodePorts that maps to the Ingress' ports 80 (http) and 443 (https).
     # nodePort service type is not created for AWS provider as LoadBalancer is used there.
     nodePorts:
+
```

Comment:
I think this should also move to the bottom. As if these settings are used the ports depend on the ELB created by aws-operator and should not be changed.

Reply:
moved it as low as possible within the object